homeassistant/components/homeassistant/exposed_entities.py

```diff
@@ -0,0 +1,351 @@
+
+        self._assistants = assistants
+
+    @callback
+    def _async_schedule_save(self) -> None:
+        """Schedule saving the preferences."""
+        self._store.async_delay_save(self._data_to_save, SAVE_DELAY)
+
+    @callback
+    def _data_to_save(self) -> dict[str, dict[str, dict[str, Any]]]:
+        """Return data to store in a file."""
+        data = {}
+
+        data["assistants"] = {
+            domain: preferences.to_json()
+            for domain, preferences in self._assistants.items()
+        }
+
+        return data
+
+
+@callback
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "homeassistant/expose_entity",
+        vol.Required("assistants"): [vol.In(KNOWN_ASSISTANTS)],
+        vol.Required("entity_ids"): [str],
+        vol.Required("should_expose"): bool,
+    }
+)
+def ws_expose_entity(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """Expose an entity to an assistant."""
+    entity_registry = er.async_get(hass)
+    entity_ids: str = msg["entity_ids"]
+
+    if blocked := next(
+        (
+            entity_id
+            for entity_id in entity_ids
+            if entity_id in CLOUD_NEVER_EXPOSED_ENTITIES
+        ),
+        None,
+    ):
+        connection.send_error(
+            msg["id"], websocket_api.const.ERR_NOT_ALLOWED, f"can't expose '{blocked}'"
+        )
+        return
+
+    if unknown := next(
+        (
+            entity_id
+            for entity_id in entity_ids
+            if entity_id not in entity_registry.entities
+        ),
+        None,
+    ):
+        connection.send_error(
+            msg["id"], websocket_api.const.ERR_NOT_FOUND, f"can't expose '{unknown}'"
+        )
+        return
+
+    exposed_entities: ExposedEntities = hass.data[DATA_EXPOSED_ENTITIES]
+    for entity_id in entity_ids:
+        for assistant in msg["assistants"]:
+            exposed_entities.async_expose_entity(
+                assistant, entity_id, msg["should_expose"]
+            )
+    connection.send_result(msg["id"])
+
+
+@callback
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "homeassistant/expose_new_entities/get",
+        vol.Required("assistant"): vol.In(KNOWN_ASSISTANTS),
+    }
+)
+def ws_expose_new_entities_get(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """Check if new entities are exposed to an assistatant."""
+    exposed_entities: ExposedEntities = hass.data[DATA_EXPOSED_ENTITIES]
+    expose_new = exposed_entities.async_get_expose_new_entities(msg["assistant"])
+    connection.send_result(msg["id"], {"expose_new": expose_new})
+
+
+@callback
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "homeassistant/expose_new_entities/set",
```

Comment:
do you think that we should keep this a bit more generic as `set_options` ?

Reply:
I'm not sure. The only other option today is 2fa override for Google which also stores data in entity options but has its own API. We can unify if it grows beyond that maybe?